src/coreclr/nativeaot/CMakeLists.txt

```diff
@@ -14,9 +14,7 @@ if(MSVC)
 endif (MSVC)
 
 if(CLR_CMAKE_HOST_UNIX)
-  if (NOT CLR_CMAKE_TARGET_ARCH_WASM)     # Native AOT runtime targeting WASM uses C++ exception handling to implement managed exceptions
-    add_compile_options(-fno-exceptions)  # Regular Native AOT runtime doesn't use C++ exception handling
-  endif()
```

Comment:
There is a little hole in the reasoning here: a couple of helpers (allocators + CID dispatch) may call out to managed code that may in turn throw, and the exception will pass through a C++ frame. It works out, however, since these exceptions can unwind through any WASM frame, however it was compiled.